components/script/script_runtime.rs

```diff
@@ -776,3 +780,44 @@ impl Deref for JSContext {
         &self.0
     }
 }
+
+pub struct StreamConsumer(*mut JSStreamConsumer);
+
+#[allow(unsafe_code)]
+impl StreamConsumer {
+    fn consumeChunk(sc: &mut JSStreamConsumer, begin: *const u8, length: usize) -> bool {
+        unsafe {
+            return StreamConsumerConsumeChunk(sc, begin, length);
+        }
+    }
+
+    fn streamEnd(sc: &mut JSStreamConsumer){
+        unsafe {
+            StreamConsumerStreamEnd(sc);
+        }
+    }
+
+    fn streamError(sc: &mut JSStreamConsumer, errorCode: usize){
+        unsafe {
+            StreamConsumerStreamError(sc, errorCode);
+        }
+    }
+
+    fn noteResponseURLs(
+        sc: &mut JSStreamConsumer,
+        maybeUrl: *const c_char,
+        maybeSourceMapUrl: *const c_char
```

Comment:
Let's use `Option<String>` arguments here. We can convert them to C strings with https://doc.rust-lang.org/stable/std/ffi/struct.CString.html, or use `ptr::null()` if there are [None](https://doc.rust-lang.org/stable/std/option/enum.Option.html) values instead of Some values present.

Reply:
Done. Please review